src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Base64StringAttribute.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+
+namespace System.ComponentModel.DataAnnotations
+{
+    /// <summary>
+    ///     Specifies that a data field value is a well-formed Base64 string.
+    /// </summary>
+    /// <remarks>
+    ///     Recognition of valid Base64 is delegated to the <see cref="Convert"/> class,
+    ///     using the <see cref="Convert.TryFromBase64String(string, Span{byte}, out int)"/> method.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = false)]
+    public class Base64StringAttribute : ValidationAttribute
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="Base64StringAttribute"/> class.
+        /// </summary>
+        public Base64StringAttribute()
+        {
+            // Set DefaultErrorMessage not ErrorMessage, allowing user to set
+            // ErrorMessageResourceType and ErrorMessageResourceName to use localized messages.
+            DefaultErrorMessage = SR.Base64StringAttribute_Invalid;
+        }
+
+        /// <summary>
+        ///     Determines whether a specified object is valid. (Overrides <see cref="ValidationAttribute.IsValid(object)" />)
+        /// </summary>
+        /// <param name="value">The object to validate.</param>
+        /// <returns>
+        ///     <see langword="true" /> if <paramref name="value"/> is <see langword="null"/> or is a valid Base64 string,
+        ///     otherwise <see langword="false" />
+        /// </returns>
+        public override bool IsValid(object? value)
+        {
+            if (value is null)
+            {
+                return true;
```

Comment:
>return true; [](http://example.com/codeflow?start=16&length=12)

Is this the expected behavior?

Reply:
Yes. Compare with other string validators like `PhoneAttribute`:

https://github.com/dotnet/runtime/blob/71034dd2fbbd2304fac5c24d3a9f764a3c65f781/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/PhoneAttribute.cs#L25-L28

The idea is that `null` validation is the responsibility of `RequiredAttribute` which has already been run, so other attributes will not try to intercept nulls.

Reply:
I wanted to echo @eiriktsarpalis's comment here. It's important for every validator to indicate a Valid state for null values so that they can be used on _optional_ properties. As was said, it's exclusively the job of `RequiredAttribute` to ensure values are present.